File: Samael/Radar/LockingRadar.cs
namespace kenran.Samael.Radar
{
    using Robocode;
    using Robocode.Util;

    internal sealed class LockingRadar
    {
        private const double LockMultiplier = 2.0d;
        private readonly AdvancedRobot bot;

        internal LockingRadar(AdvancedRobot bot)
        {
            this.bot = bot;
        }

        internal void OnScannedRobot(ScannedRobotEvent evnt)
        {
            double radarTurn = Utils.NormalRelativeAngle(bot.HeadingRadians + evnt.BearingRadians - bot.RadarHeadingRadians);
            bot.SetTurnRadarRightRadians(LockMultiplier * radarTurn);
        }
    }
}

File: Oldur/src/kenran/radar/LockingRadar.kt
package kenran.radar

import robocode.AdvancedRobot
import robocode.ScannedRobotEvent
import robocode.util.Utils

class LockingRadar(bot: AdvancedRobot, lockMultiplier: Double) {
	private val _bot = bot
    private val _lockMultiplier = lockMultiplier

    fun onScannedRobot(e: ScannedRobotEvent) {
		val angleToEnemy = _bot.headingRadians + e.bearingRadians - _bot.radarHeadingRadians
		val radarTurn = Utils.normalRelativeAngle(angleToEnemy)
		_bot.setTurnRadarRightRadians(_lockMultiplier * radarTurn)
	}
}